Framework/Core/src/runDataProcessing.cxx

```diff
@@ -311,20 +325,32 @@ int createPipes(int maxFd, int *pipes) {
     return maxFd;
 }
 
-// Kill all the active children
-void killChildren(std::vector<DeviceInfo> &infos) {
+// Kill all the active children. Exit code
+// is != 0 if any of the children had an error.
+int killChildren(std::vector<DeviceInfo> &infos) {
+  int exitCode = 0;
   for (auto &info : infos) {
+    if (exitCode == 0
+        && info.maxLogLevel >= LogParsingHelpers::LogLevel::Error){
+      exitCode = 1;
+    }
```

Comment:
actually, why use return codes for error conditions? would the use of exceptions not be clearer? This I mean in general, i have seen quite some code in O2 already that uses retCodes and/or assert() - should we not try to push it in a consistent direction?

Reply:
This is not an error condition. It's the calculated value which the driver will use to call `exit()`. So I would say it does not apply here. 

That said, I would personally avoid at all costs exceptions for anything but things which requires exiting the computation immediately and discarding the data.